tests/test_registries.py

```diff
@@ -0,0 +1,166 @@
+#
+# (c) Copyright 2017-2018 Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+
+import unittest
+
+import fixtures
+import testtools
+
+from . import base  # noqa: H304
+
+import windlass.registries
+
+
+class TestRegistries(testtools.TestCase):
+
+    def test_from_url_aws_ecr(self):
+        registry_url = "000000000000.dkr.ecr.us-west-1.amazonaws.com"
+        registry = windlass.registries.from_url(registry_url)
+
+        self.assertIsInstance(registry, windlass.registries.EcrDockerRegistry)
+
+    def test_from_url_jfrog(self):
+        registry_url = "example-org.jfrog.io"
+        registry = windlass.registries.from_url(registry_url)
+
+        self.assertIsInstance(
+            registry, windlass.registries.JfrogDockerRegistry
+        )
+
+    def test_from_url_hub(self):
+        registry_url = "registry.hub.docker.com"
+        registry = windlass.registries.from_url(registry_url)
+
+        self.assertIsInstance(registry, windlass.registries.DockerRegistry)
+
+    def test_from_url_other(self):
+        registry_url = "registry.quay.io"
+        registry = windlass.registries.from_url(registry_url)
+
+        self.assertIsInstance(registry, windlass.registries.DockerRegistry)
+
+
+class TestDockerRegistry(testtools.TestCase):
+
+    reg_url = "registry.hub.docker.com"
+
+    @base.addfixture(fixtures.EnvironmentVariable, 'DOCKER_USER', 'my_user')
```

Comment:
Lines 69-70 are identical - this could be replaced with setUp where useFixture is used and it will be called separetaly for each test.

Reply:
Using `self.useFixture(...)`